ios/brave-ios/Sources/BraveVPN/Components/BuyVPN/Paywall/BraveVPNPaywallView.swift

```diff
@@ -12,111 +12,52 @@ import StoreKit
 import SwiftUI
 import os.log
 
-enum BraveVPNPaymentStatus {
-  case ongoing
-  case success
-  case failure
-}
-
 public struct BraveVPNPaywallView: View {
-  @Environment(\.dismiss)
-  private var dismiss
-
-  @State
-  private var selectedTierType: BraveVPNSubscriptionTier = .monthly
-
-  @State
-  private var availableTierTypes: [BraveVPNSubscriptionTier] = [.yearly, .monthly]
-
-  @State
-  private var paymentStatus: BraveVPNPaymentStatus = .success
-
-  @State
-  private var isShowingPurchaseAlert = false
-
-  @State
-  private var shouldRefreshCredentials = false
-
-  @State
-  private var shouldRedeemPromoCode = false
-
-  @State
-  private var isFreeTrialAvailable = !Preferences.VPN.freeTrialUsed.value
+  @ObservedObject var iapObserverManager: BraveVPNIAPObserverManager
 
-  @State
-  private var iapRestoreTimer: Task<Void, Error>?
+  private let openVPNAuthenticationInNewTab: () -> Void
+  private let installVPNProfile: () -> Void
 
-  @ObservedObject
-  var iapObserverManager: BraveVPNIAPObserverManager
+  @State private var selectedTierType: BraveVPNSubscriptionTier = .yearly
+  @State private var availableTierTypes: [BraveVPNSubscriptionTier] = [.yearly, .monthly]
+  @State private var isShowingPurchaseAlert = false
+  @State private var isFreeTrialAvailable = !Preferences.VPN.freeTrialUsed.value
+  @State private var iapRestoreTimer: Task<Void, Error>?
+  @State private var orientation = UIDeviceOrientation.unknown
```

Comment:
Is it not possible to use https://developer.apple.com/documentation/swiftui/environmentvalues/horizontalsizeclass